src/abstract-interpretation/data-frame/shape-inference.ts

```diff
@@ -58,18 +58,18 @@ export function resolveIdToDataFrameShape(
 
 	if(dfg === undefined || node === undefined || domain === undefined) {
 		return;
-	} else if(domain.has(node.info.id)) {
-		return domain.get(node.info.id);
+	} else if(domain.value.has(node.info.id)) {
+		return domain.value.get(node.info.id);
 	}
 	const vertex = dfg.getVertex(node.info.id);
 	const call = vertex?.tag === VertexType.FunctionCall ? vertex : undefined;
 	const origins = Array.isArray(call?.origin) ? call.origin : [];
 
 	if(node.type === RType.Symbol) {
-		const values = getVariableOrigins(node.info.id, dfg).map(origin => domain.get(origin.info.id));
+		const values = getVariableOrigins(node.info.id, dfg).map(origin => domain.value.get(origin.info.id));
```

Comment:
I would prefer having this as a delegate, again to avoid the value map being exposed